File: Data Structures & Algorithms/count-connected-components/submission-0.go
func countComponents(n int, edges [][]int) int {
    g := make([][]int, n)
	for i:=0;i<n;i+=1{
		g[i] = make([]int, 0)
	}

	for _, e := range edges {
		u, v := e[0], e[1]
		g[u] = append(g[u], v)
		g[v] = append(g[v], u)
	}

	visited := make([]int, n)
	ans := 0
	for i:=0;i<n;i+=1{
		if visited[i] == 0{
			ans += 1
			dfs(g,i,visited)
		}
	}
	return ans
}

func dfs(g [][]int, v int, visited []int) {
	visited[v] = 1
	for i:=0;i<len(g[v]);i+=1{
		if visited[g[v][i]] == 0 {
			dfs(g,g[v][i], visited)
		}
	}
}

File: Data Structures & Algorithms/network-delay-time/submission-0.py
class Solution:
    def networkDelayTime(self, times: List[List[int]], n: int, k: int) -> int:
        g = [[] for i in range(0,n+1)]
        for u,v,t in times:
            g[u].append((v,t))
        h = [(0,k)]
        visited = [False]*(n+1)
        ans = None
        while len(h) > 0:
            dist, u  = heapq.heappop(h)
            
            if visited[u]:
                continue
            else:
                visited[u] = True
            if ans is None or ans < dist:
                ans = dist
            for v,t in g[u]:
                if not visited[v]:
                    heapq.heappush(h,(dist+t, v))
        for i in range(1,n+1):
            if not visited[i]:
                return -1
        return ans
        